radicle-keystore/src/lib.rs

```diff
@@ -0,0 +1,56 @@
+#![allow(clippy::type_complexity)]
+
+use std::convert::Infallible;
+
+use secstr::{SecStr, SecUtf8};
+
+mod crypto;
+pub mod file;
+pub mod memory;
+
+pub use file::FileStorage;
+pub use memory::MemoryStorage;
+
+pub trait Pinentry {
+    type Error: std::error::Error;
+
+    fn get_passphrase(&self) -> Result<SecUtf8, Self::Error>;
+}
+
+impl Pinentry for SecUtf8 {
+    type Error = Infallible;
+
+    fn get_passphrase(&self) -> Result<SecUtf8, Infallible> {
+        Ok(self.clone())
+    }
+}
+
+pub struct Keypair<PK, SK> {
+    pub public_key: PK,
+    pub secret_key: SK,
+}
+
+pub trait IntoSecretKey<M> {
+    fn into_secret_key(bytes: SecStr, metadata: &M) -> Self;
+}
+
+pub trait HasMetadata<M> {
```

Comment:
Yeah it doesn't really fit: `IntoSecretKey` was the inverse of `From`, but `Into` wouldn't cut it because we need the creation timestamp eventually to pull a PGP key out of thin air. I moved things into a single trait now, but then the `From` bound is still kinda dangling.